docs/rules/no-undef.md

```diff
@@ -54,6 +58,23 @@ if (typeof UndefinedIdentifier === "undefined") {
 }
 ```
 
+#### typeof
+
+You can use this option if you want to prevent `typeof` check on a variable which does not exists or is not defined.
```

Comment:
What's the difference between not existing and not being defined? If there's no difference, saying both only serves to confuse. I would say "has not been declared" or "is not in lexical scope".